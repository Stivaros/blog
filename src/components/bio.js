/**
 * Bio component that queries for data
 * with Gatsby's useStaticQuery component
 *
 * See: https://www.gatsbyjs.org/docs/use-static-query/
 */

import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import Image from "gatsby-image"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { faTwitter, faLinkedinIn } from "@fortawesome/free-brands-svg-icons"
import { faHome } from "@fortawesome/free-solid-svg-icons"

import { rhythm } from "../utils/typography"

const Bio = () => {
  const data = useStaticQuery(graphql`
    query BioQuery {
      avatar: file(absolutePath: { regex: "/profile-pic.jpg/" }) {
        childImageSharp {
          fixed(width: 50, height: 50) {
            ...GatsbyImageSharpFixed
          }
        }
      }
      site {
        siteMetadata {
          author {
            name
            summary
          }
          social {
            twitter
            linkedIn
          }
        }
      }
    }
  `)

  const { author, social } = data.site.siteMetadata
  return (
    <div
      style={{
        display: `flex`,
      }}
    >
      <Image
        fixed={data.avatar.childImageSharp.fixed}
        alt={author.name}
        style={{
          marginRight: rhythm(1 / 2),
          marginBottom: 0,
          minWidth: 50,
          borderRadius: `100%`,
        }}
        imgStyle={{
          borderRadius: `50%`,
        }}
      />
      <p>
        I am <strong>{author.name}</strong>, {author.summary}<br /><br />
        Find me online
        {` `}
        <span
          style={{
            backgroundColor: `#C5E1A5`,
            padding: `0.4em 0.25em 0.25em`,
            borderRadius: `0.25em`
          }}
        >
          <a href={`http://stivaros.com/`} style={{margin: `0 0.25em`}} aria-label="Home">
            <FontAwesomeIcon icon={faHome} />
          </a>
          <a href={`https://twitter.com/${social.twitter}`} style={{margin: `0 0.25em`}} aria-label="Twitter">
            <FontAwesomeIcon icon={faTwitter} />
          </a>
          <a href={`https://linkedin.com/in/${social.linkedIn}`} style={{margin: `0 0.25em`}} aria-label="LinkedIn">
            <FontAwesomeIcon icon={faLinkedinIn} />
          </a>
        </span>
      </p>
    </div>
  )
}

export default Bio
