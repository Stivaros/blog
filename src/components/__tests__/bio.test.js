import React from "react"
import { render, screen } from "@testing-library/react"
import { PureBio, Bio } from "../bio"

describe("Bio", () => {
  const defaultData = {
    avatar: {
      childImageSharp: {
        gatsbyImageData: {
          layout: 'fixed',
          width: 50,
          height: 50,
          images: {
            fallback: {
              src: 'test-image.jpg'
            }
          }
        }
      },
    },
    site: {
      siteMetadata: {
        author: {
          name: 'Stath',
          summary: 'A test blog',
        },
        social: {
          twitter: 'twitterHandle',
          linkedIn: 'linkedInHandle',
        },
      },
    },
  }

  describe("PureBio", () => {
    it("renders author information with avatar", () => {
      render(<PureBio data={defaultData} />)
  
      const { name, summary } = defaultData.site.siteMetadata.author
      
      const avatar = screen.getByTestId('gatsby-image')
      expect(avatar).toHaveAttribute('alt', name)
      
      expect(screen.getByText(new RegExp(name))).toBeInTheDocument()
      expect(screen.getByText(new RegExp(summary))).toBeInTheDocument()
    })
  
    describe("social links", () => {
      beforeEach(() => {
        render(<PureBio data={defaultData} />)
      })
  
      it("renders home link", () => {
        const homeLink = screen.getByRole('link', { name: /home/i })
        expect(homeLink).toHaveAttribute('href', 'http://stivaros.com/')
      })
  
      it("renders Twitter link", () => {
        const { twitter } = defaultData.site.siteMetadata.social
        const twitterLink = screen.getByRole('link', { name: /twitter/i })
        expect(twitterLink).toHaveAttribute('href', `https://twitter.com/${twitter}`)
      })
  
      it("renders LinkedIn link", () => {
        const { linkedIn } = defaultData.site.siteMetadata.social
        const linkedInLink = screen.getByRole('link', { name: /linkedin/i })
        expect(linkedInLink).toHaveAttribute('href', `https://linkedin.com/in/${linkedIn}`)
      })
  
      it("groups social links together", () => {
        const socialContainer = screen.getByLabelText(/social links/i)
        expect(socialContainer).toBeInTheDocument()
      })
    })
  })

  describe("Bio with GraphQL", () => {
    it("renders using data from useStaticQuery", () => {
      render(<Bio />)

      expect(screen.getByText(/Stath/)).toBeInTheDocument()
      expect(screen.getByText(/A test blog/)).toBeInTheDocument()
      expect(screen.getByRole('link', { name: /twitter/i })).toHaveAttribute('href', 'https://twitter.com/twitterHandle')
      expect(screen.getByRole('link', { name: /linkedin/i })).toHaveAttribute('href', 'https://linkedin.com/in/linkedInHandle')
    })
  })
})