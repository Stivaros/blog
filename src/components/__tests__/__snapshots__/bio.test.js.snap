// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <p>
    I am 
    <strong>
      Stath
    </strong>
    , 
    A test blog
    <br />
    <br />
    Find me online
     
    <span
      style={
        Object {
          "backgroundColor": "#C5E1A5",
          "borderRadius": "0.25em",
          "padding": "0.4em 0.25em 0.25em",
        }
      }
    >
      <a
        aria-label="Home"
        href="http://stivaros.com/"
        style={
          Object {
            "margin": "0 0.25em",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-home "
          data-icon="home"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
      <a
        aria-label="Twitter"
        href="https://twitter.com/twitterHandle"
        style={
          Object {
            "margin": "0 0.25em",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-twitter "
          data-icon="twitter"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
      <a
        aria-label="LinkedIn"
        href="https://linkedin.com/in/linkedInHandle"
        style={
          Object {
            "margin": "0 0.25em",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-linkedin-in "
          data-icon="linkedin-in"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
    </span>
  </p>
</div>
`;
